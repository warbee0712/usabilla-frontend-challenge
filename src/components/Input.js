import styled from 'styled-components'
import { color, space, fontSize, borderRadius, width } from 'styled-system'
import { focus } from '../sharedStyles'

const Input = styled.input`
  border: 1px solid ${props => props.theme.colors.light3};
  ${borderRadius}
  ${color}
  ${fontSize}
  ${space}
  ${width}
  &:hover {
    box-shadow: ${props => props.theme.shadows.default};
  }
  &:focus {
    ${focus}
  }
`

Input.propTypes = {
  ...borderRadius.propTypes,
  ...color.propTypes,
  ...fontSize.propTypes,
  ...space.propTypes,
  ...width.propTypes,
}

Input.defaultProps = {
  borderRadius: 'default',
  color: 'dark3',
  bg: 'white',
  fontSize: 'xs',
  px: 'm',
  py: 's',
}

export default Input
