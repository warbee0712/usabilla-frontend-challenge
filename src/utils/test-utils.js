import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import theme from '../theme'

const customRender = (node, options) => {
  return render(<ThemeProvider theme={theme}>{node}</ThemeProvider>, options)
}

// re-export everything
export * from 'react-testing-library'

// override render method
export { customRender as render }
