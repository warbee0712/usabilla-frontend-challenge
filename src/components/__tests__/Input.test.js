import React from 'react'
import { render } from 'test-utils'
import Input from '../Input'

describe('Input', () => {
  it('should match snapshot', () => {
    const { container } = render(<Input placeholder="Search here..." />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
