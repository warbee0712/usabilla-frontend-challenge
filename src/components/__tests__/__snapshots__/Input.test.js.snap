// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
.c0 {
  border: 1px solid #c8d2da;
  border-radius: 2px;
  color: #59636b;
  background-color: #fff;
  font-size: 0.8125rem;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.6875rem;
  padding-bottom: 0.6875rem;
}

.c0:hover {
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
}

.c0:focus {
  border: 1px solid #00a5c9;
  box-shadow: 0 1px 2px rgba(0,0,0,0.15);
  outline: none;
}

<input
  class="c0"
  color="dark3"
  font-size="xs"
  placeholder="Search here..."
/>
`;
